File: backend/app/contracts/artifacts.py
from __future__ import annotations

from datetime import datetime

from app.contracts.common import JsonValue, StrictModel


class ArtifactMetadata(StrictModel):
    artifact_ref: str
    workflow_id: str
    ticket_id: str
    node_id: str
    path: str
    kind: str
    media_type: str | None = None
    status: str
    materialization_status: str
    lifecycle_status: str
    retention_class: str
    retention_class_source: str | None = None
    retention_ttl_sec: int | None = None
    retention_policy_source: str | None = None
    expires_at: datetime | None = None
    deleted_at: datetime | None = None
    deleted_by: str | None = None
    delete_reason: str | None = None
    storage_deleted_at: datetime | None = None
    size_bytes: int | None = None
    content_hash: str | None = None
    created_at: datetime
    content_url: str
    download_url: str
    preview_url: str


class ArtifactMetadataEnvelope(StrictModel):
    data: ArtifactMetadata


class ArtifactPreviewData(StrictModel):
    artifact_ref: str
    preview_kind: str
    media_type: str | None = None
    lifecycle_status: str
    content_url: str | None = None
    download_url: str | None = None
    json_content: JsonValue | None = None
    text_content: str | None = None


class ArtifactPreviewEnvelope(StrictModel):
    data: ArtifactPreviewData

File: frontend/src/types/artifacts.ts
export type ArtifactMetadata = {
  artifact_ref: string
  workflow_id: string
  ticket_id: string
  node_id: string
  path: string
  kind: string
  media_type: string | null
  preview_kind: string | null
  display_hint: string | null
  status: string
  materialization_status: string
  lifecycle_status: string
  retention_class: string
  retention_class_source: string | null
  retention_ttl_sec: number | null
  retention_policy_source: string | null
  expires_at: string | null
  deleted_at: string | null
  deleted_by: string | null
  delete_reason: string | null
  storage_backend: string
  storage_object_key: string | null
  storage_delete_status: string
  storage_delete_error: string | null
  storage_deleted_at: string | null
  size_bytes: number | null
  content_hash: string | null
  created_at: string
  content_url: string
  download_url: string
  preview_url: string
}

export type ArtifactMetadataEnvelope = {
  data: ArtifactMetadata
}

export type ArtifactPreview = {
  artifact_ref: string
  preview_kind: string
  media_type: string | null
  lifecycle_status: string
  content_url: string | null
  download_url: string | null
  json_content: unknown
  text_content: string | null
}

export type ArtifactPreviewEnvelope = {
  data: ArtifactPreview
}
